pkg/jobs/jobsprotectedts/jobs_protected_ts_test.go

```diff
@@ -96,10 +96,13 @@ func TestJobsProtectedTimestamp(t *testing.T) {
 	_, recRemains := mkJobAndRecord()
 	ensureNotExists := func(ctx context.Context, txn *kv.Txn, ptsID uuid.UUID) (err error) {
 		_, err = ptp.GetRecord(ctx, txn, ptsID)
+		if err == nil {
+			return errors.New("found pts record, waiting for ErrNotExists")
```

Comment:
I'm not sure if these should be assertion errors instead? I'm leaning towards no based on my understanding of assertion errors being used when we think a situation is a programmatic error, or violation of an invariant.

Reply:
Agreed, this error is expected.